src/graph/decremental_connectivity.rs

```diff
@@ -0,0 +1,228 @@
+use std::collections::HashSet;
+
+/// A data-structure that, given a forest, allows dynamic-connectivity queries.
+/// Meaning deletion of an edge (u,v) and checking whether two vertecies are still connected.
+///
+/// # Complexity
+/// The preprocessing phase runs in O(n) time, where n is the the number of vertecies in the forest.
+/// Deletion runs in O(log n) and checking for connectivity runs in O(1) time.
+///
+/// # Sources
+/// used Wikipedia as reference: <https://en.wikipedia.org/wiki/Dynamic_connectivity>
+pub struct DecrementalConnectivity {
+    adjacent: Vec<HashSet<usize>>,
```

Comment:
I think you are assuming that your graph is not directed, i.e. each edge is in two directions. Is there a check somewhere that if `self.adjacent[a].contains(&b)`, then `self.adjacent[b].contains(&a)`?

Reply:
added a check in `has_cycle()`